cmd/src/lsif_upload.go

```diff
@@ -36,28 +39,84 @@ Examples:
 		fmt.Println(usage)
 	}
 	var (
-		repoFlag        = flagSet.String("repo", "", `The name of the repository. (required)`)
-		commitFlag      = flagSet.String("commit", "", `The 40-character hash of the commit. (required)`)
-		fileFlag        = flagSet.String("file", "", `The path to the LSIF dump file. (required)`)
+		repoFlag        = flagSet.String("repo", "", `The name of the repository. By default, derived from the origin remote.`)
+		commitFlag      = flagSet.String("commit", "", `The 40-character hash of the commit. Defaults to the currently checked-out commit.`)
+		fileFlag        = flagSet.String("file", "./dump.lsif", `The path to the LSIF dump file.`)
 		githubTokenFlag = flagSet.String("github-token", "", `A GitHub access token with 'public_repo' scope that Sourcegraph uses to verify you have access to the repository.`)
-		rootFlag        = flagSet.String("root", "", `The path in the repository that matches the LSIF projectRoot (e.g. cmd/)`)
+		rootFlag        = flagSet.String("root", "", `The path in the repository that matches the LSIF projectRoot (e.g. cmd/project1). Defaults to the empty string, which refers to the top level of the repository.`)
 		apiFlags        = newAPIFlags(flagSet)
 	)
 
 	handler := func(args []string) error {
 		flagSet.Parse(args)
 
-		ensureSet := func(value *string, argName string) {
-			if value == nil || *value == "" {
-				fmt.Printf("src lsif: no %s supplied\n", argName)
-				fmt.Printf("Run 'src lsif help' for usage.\n")
+		if repoFlag == nil || *repoFlag == "" {
+			remoteURL, err := exec.Command("git", "remote", "get-url", "origin").Output()
+			if err != nil {
+				fmt.Println(err)
+				fmt.Println("Unable to detect repository from environment.")
+				fmt.Println("Either cd into a git repository or set -repo explicitly.")
+				os.Exit(1)
+			}
+			*repoFlag, err = parseRemoteURL(strings.TrimSpace(string(remoteURL)))
+			if err != nil {
+				fmt.Println(err)
+				fmt.Println("Set -repo explicitly.")
 				os.Exit(1)
 			}
 		}
+		fmt.Println("Repository: " + *repoFlag)
```

Comment:
Do we want these to print or was it part of debugging?

Reply:
I'm open to feedback here. Printing them makes the defaults transparent.